contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >= 0.7.0 < 0.9.0;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENSRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@ew-did-registry/proxyidentity/contracts/IOwned.sol";
+import "./RoleDefinitionResolver.sol";
+import {ClaimManager} from "./ClaimManager.sol";
+
+interface EthereumDIDRegistry {
+    function identityOwner(address identity) external view returns(address);
+    function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool);
+}
+    
+contract RevocationRegistryOnChain {
+    bytes32 constant private ASSERTION_DELEGATE_TYPE = 0x766572694b657900000000000000000000000000000000000000000000000000;
+    string constant private ERC712_DOMAIN_NAME = "Revocation Registry";
+    string constant private ERC712_DOMAIN_VERSION = "1.0";
+
+    struct RevokedClaim {
+        address revoker;
+        uint revokedTimestamp;
+        bytes32 revokerRoleDigest; //namehash
+    }
+
+    mapping(bytes32 => RevokedClaim) revokedClaims;
+    address private didRegistry;
+    address private ensRegistry;
+    address private claimManager;
+    
+    event Revoked(address revoker, bytes32 claimDigest);
+
+    constructor(address _didRegistry, address _ensRegistry, address _claimManager){
+        didRegistry = _didRegistry;
+        ensRegistry = _ensRegistry;
+        claimManager = _claimManager;
+    }
+
+    function revokeClaim(bytes32 claimDigest, bytes32 role, address revoker, bytes32 revokerRoleDigest) public {
+        require(revokedClaims[claimDigest].revokedTimestamp == 0, "The claim is already revoked");
+        verifyRevoker(revoker, role, revokerRoleDigest);
+        revokedClaims[claimDigest].revoker = revoker;
+        revokedClaims[claimDigest].revokedTimestamp = block.number;
```

Comment:
if you used `block.number` here on purpose as the revokedTimestamp, what is the reason for that ? I would expect to find  
```suggestion
        revokedClaims[claimDigest].revokedTimestamp = block.timestamp;
```

Reply:
block.timestamp can be influenced by the miners, though the manipulation would be minimal and rare but a possibility exist. It is recommended to use block.number, let me know your thoughts.